gpflow/models/sgpr.py

```diff
@@ -59,34 +59,32 @@ def upper_bound(self):
         kuu = Kuu(self.inducing_variable, self.kernel, jitter=default_jitter())
         kuf = Kuf(self.inducing_variable, self.kernel, x_data)
 
+        KufKfu = tf.linalg.matmul(kuf, kuf, transpose_b=True)
+
         L = tf.linalg.cholesky(kuu)
-        LB = tf.linalg.cholesky(kuu + self.likelihood.variance ** -1.0 *
-                                tf.linalg.matmul(kuf, kuf, transpose_b=True))
+        LB = tf.linalg.cholesky(kuu + KufKfu / self.likelihood.variance)
 
-        Linvkuf = tf.linalg.triangular_solve(L, kuf, lower=True)
+        LinvKuf = tf.linalg.triangular_solve(L, kuf, lower=True)
         # Using the Trace bound, from Titsias' presentation
-        # c = tf.reduce_sum(Kdiag) - tf.reduce_sum(Linvkuf ** 2.0)
-        Kff = self.kernel(x_data)
-        Qff = tf.linalg.matmul(kuf, Linvkuf, transpose_a=True)
+        c = tf.reduce_sum(Kdiag) - tf.reduce_sum(tf.square(LinvKuf))
+        # Kff = self.kernel(x_data)
```

Comment:
Can you remove it, if you don't need it anymore
```suggestion
```

Reply:
I'm only reflecting the code that was in there in GPflow 1 - all I did was trying to make it work in gpflow2. I don't know if it may be needed in the future - I assume not - but I'll let someone else make that decision, and maybe better in a separate PR.